tests/components/iaqualink/test_config_flow.py

```diff
@@ -0,0 +1,52 @@
+"""Tests for iAqualink config flow."""
+from unittest.mock import patch
+
+import iaqualink
+
+from homeassistant.components.iaqualink import config_flow
+
+from tests.common import MockConfigEntry, mock_coro
+
+DATA = {"username": "test@example.com", "password": "pass"}
+
+
+async def test_already_configured(hass):
+    """Test if a HomeKit discovered bridge has already been configured."""
+    MockConfigEntry(domain="iaqualink", data=DATA).add_to_hass(hass)
+
+    flow = config_flow.AqualinkFlowHandler()
+    flow.hass = hass
+    flow.context = {}
+
+    result = await flow.async_step_user(DATA)
+
+    assert result["type"] == "abort"
+
+
+async def test_import_with_invalid_credentials(hass):
+    """Test config flow ."""
+    flow = config_flow.AqualinkFlowHandler()
+    flow.hass = hass
+
+    with patch(
+        "iaqualink.AqualinkClient.login", side_effect=iaqualink.AqualinkLoginException
+    ):
+        result = await flow.async_step_user(DATA)
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "connection_failure"}
+
+
+async def test_import_with_existing_config(hass):
```

Comment:
This isn't what the test does. We should test both the user step and the import step to completion.

Reply:
I get renaming this test since it actually calls async_step_user() but Is there a point of having separate tests for async_step_import() and async_step_user() since import does nothing besides returning async_step_user()?

Reply:
All code in the module needs to be tested.

Reply:
How would you know it works if you are not testing it 😄

Reply:
Do I need to use pytest.mark.parametrize() to duplicate each of the 3 tests?